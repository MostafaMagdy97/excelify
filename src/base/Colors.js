const baseColors = {
  primary: '#0070E0',
  secondary: '#06BEE1',
  success: '#44AF69',
  warning: '#FCAB10',
  danger: '#F8333C',
  black: '#1b2733',
  grey: '#637282',
  light: '#f5f5f5',
  white: '#fff',
  transparent: 'transparent',
};

const COLORS = {
  BLUE: 'blue',
  RED: 'red',
  DANGER: 'danger',
  PAGE_BACKGROUND: 'pageBackground',
  SUCCESS: 'success',
  WARNING: 'warning',
  LINK: 'link',
  DISABLED: 'disabled',
  TEXT: 'text',
  HELP_TEXT: 'helpText',
  BORDER: 'border',
  HEADER: 'header',
  WHITE: 'white',
  SEPARATOR: 'separator',
  TRANSPARENT: 'transparent',
};

const COLORS_VALUES = {
  [COLORS.BLUE]: baseColors.primary,
  [COLORS.PRIMARY_RED]: baseColors.secondary,
  [COLORS.DANGER]: baseColors.danger,
  [COLORS.PAGE_BACKGROUND]: baseColors.light,
  [COLORS.SUCCESS]: baseColors.success,
  [COLORS.WARNING]: baseColors.warning,
  [COLORS.LINK]: baseColors.primary,
  [COLORS.DISABLED]: baseColors.grey,
  [COLORS.TEXT]: baseColors.black,
  [COLORS.HELP_TEXT]: baseColors.grey,
  [COLORS.BORDER]: baseColors.light,
  [COLORS.HEADER]: baseColors.primary,
  [COLORS.WHITE]: baseColors.white,
  [COLORS.SEPARATOR]: baseColors.grey,
};

export { COLORS, COLORS_VALUES };
export default COLORS;
