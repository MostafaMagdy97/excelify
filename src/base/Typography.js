export const FONT_TYPES = {
  HERO_TITLE: 'heroTitle',
  SUPER_TITLE: 'superTitle',
  TITLE: 'title',
  HEADING: 'heading',
  SUBHEADING: 'subheading',
  BODY: 'body',
  CAPTION: 'caption',
};

export const FONT_SIZES = {
  // Breakpoints: ['400px', '576px', '786px', '992px']
  [FONT_TYPES.HERO_TITLE]: ['42px', '42px', '42x', '60px'],
  [FONT_TYPES.SUPER_TITLE]: ['32px', '32px', '32x', '30px'],
  [FONT_TYPES.TITLE]: ['21px', '21px', '20px', '20px'],
  [FONT_TYPES.HEADING]: ['19px', '19px', '18px', '18px'],
  [FONT_TYPES.SUBHEADING]: ['17px', '17px', '16px', '16px'],
  [FONT_TYPES.BODY]: ['15px', '15px', '14px', '14px'],
  [FONT_TYPES.CAPTION]: ['13px', '13px', '12px', '12px'],
};

export const FONT_WEIGHTS = {
  SEMI_BOLD: 600,
  NORMAL: 500,
  LIGHT: 400,
  SUPER_LIGHT: 300,
};

export const LINE_HEIGHTS = {
  [FONT_TYPES.HERO_TITLE]: '30px',
  [FONT_TYPES.SUPER_TITLE]: '30px',
  [FONT_TYPES.TITLE]: '24px',
  [FONT_TYPES.HEADING]: '24px',
  [FONT_TYPES.SUBHEADING]: '24px',
  [FONT_TYPES.BODY]: '24px',
  [FONT_TYPES.CAPTION]: '24px',
};
