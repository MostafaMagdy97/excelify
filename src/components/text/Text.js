import * as React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { space, width, fontSize, color, fontWeight, lineHeight, zIndex } from 'styled-system';

import { FONT_SIZES, FONT_WEIGHTS, FONT_TYPES, LINE_HEIGHTS } from '../../base/Typography';
import { COLORS } from '../../base/Colors';

const propTypes = {
  fontWeight: PropTypes.oneOf(Object.values(FONT_WEIGHTS)),
  fontSize: PropTypes.oneOf(Object.values(FONT_SIZES)),
  lineHeight: PropTypes.oneOf(Object.values(LINE_HEIGHTS)),
  color: PropTypes.oneOf(Object.values(COLORS)),
  tag: PropTypes.string,
  children: PropTypes.string.isRequired,
};

/**
 * Default props will render <label type="body" />
 */
const defaultProps = {
  fontWeight: FONT_WEIGHTS.NORMAL,
  fontSize: FONT_SIZES[FONT_TYPES.BODY],
  lineHeight: LINE_HEIGHTS[FONT_TYPES.BODY],
  color: COLORS.TEXT,
  tag: 'label',
};

/**
 * Generate basic HTML node
 * @param {object} props
 */
const Base = props => {
  const Tag = props.tag;
  return <Tag {...props}>{props.children}</Tag>;
};

Base.propTypes = propTypes;
Base.defaultProps = defaultProps;

const Text = styled(Base)`
  ${space}
  ${width}
  ${color}
  ${fontWeight}
  ${fontSize}
  ${lineHeight}
  ${zIndex}
  font-display: fallback;
`;

Text.propTypes = propTypes;
Text.defaultProps = defaultProps;

const TextHOC = props => {
  const { type } = props;
  return <Text {...props} fontSize={FONT_SIZES[type]} lineHeight={LINE_HEIGHTS[type]} />;
};

TextHOC.propTypes = {
  type: PropTypes.oneOf(Object.values(FONT_TYPES)),
};

TextHOC.defaultProps = {
  type: FONT_TYPES.BODY,
};

export default TextHOC;
