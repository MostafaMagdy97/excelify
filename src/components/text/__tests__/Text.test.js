import React from 'react';
import { configure, render, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';

import Text from '../Text';
import theme from '../../../base/Theme';
import { FONT_WEIGHTS, FONT_TYPES, LINE_HEIGHTS } from '../../../base/Typography';
import { COLORS } from '../../../base/Colors';

configure({ adapter: new Adapter() });

describe('Typography', () => {
  it('should render', () => {
    const component = (
      <ThemeProvider theme={theme}>
        <Text>Some sort of text</Text>
      </ThemeProvider>
    );
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });

  it('should render type correctly', () => {
    const component = (
      <ThemeProvider theme={theme}>
        <Text type={FONT_TYPES.heading}>Some sort of text</Text>
      </ThemeProvider>
    );
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });

  it('should render font weight correctly', () => {
    const component = (
      <ThemeProvider theme={theme}>
        <Text fontWeight={FONT_WEIGHTS.semiBold}>Some sort of text</Text>
      </ThemeProvider>
    );
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });

  it('should render color correctly', () => {
    const component = (
      <ThemeProvider theme={theme}>
        <Text color={COLORS.TEXT}>Some sort of text</Text>
      </ThemeProvider>
    );
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });

  it('should render line height correctly', () => {
    const component = (
      <ThemeProvider theme={theme}>
        <Text lineHeight={LINE_HEIGHTS[FONT_TYPES.BODY]}>Some sort of text</Text>
      </ThemeProvider>
    );
    const tree = render(component);
    expect(tree).toMatchSnapshot();
  });

  it('should render tag correctly', () => {
    const component = (
      <ThemeProvider theme={theme}>
        <Text tag="h1">Some sort of text</Text>
      </ThemeProvider>
    );
    const tree = shallow(component)
      .dive()
      .dive()
      .dive();
    expect(tree.type().toString()).toEqual('h1');
  });
});
