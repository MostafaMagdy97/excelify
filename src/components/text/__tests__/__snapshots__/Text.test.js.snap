// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render 1`] = `
.c0 {
  color: #1b2733;
  font-weight: 500;
  font-size: 15px;
  font-display: fallback;
}

@media screen and (min-width:400px) {
  .c0 {
    font-size: 15px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    font-size: 14px;
  }
}

@media screen and (min-width:786px) {
  .c0 {
    font-size: 14px;
  }
}

<label
  class="c0"
  color="text"
  font-size="15px,15px,14px,14px"
  font-weight="500"
  tag="label"
  type="body"
>
  Some sort of text
</label>
`;

exports[`Typography should render color correctly 1`] = `
.c0 {
  color: #1b2733;
  font-weight: 500;
  font-size: 15px;
  font-display: fallback;
}

@media screen and (min-width:400px) {
  .c0 {
    font-size: 15px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    font-size: 14px;
  }
}

@media screen and (min-width:786px) {
  .c0 {
    font-size: 14px;
  }
}

<label
  class="c0"
  color="text"
  font-size="15px,15px,14px,14px"
  font-weight="500"
  tag="label"
  type="body"
>
  Some sort of text
</label>
`;

exports[`Typography should render font weight correctly 1`] = `
.c0 {
  color: #1b2733;
  font-weight: 500;
  font-size: 15px;
  font-display: fallback;
}

@media screen and (min-width:400px) {
  .c0 {
    font-size: 15px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    font-size: 14px;
  }
}

@media screen and (min-width:786px) {
  .c0 {
    font-size: 14px;
  }
}

<label
  class="c0"
  color="text"
  font-size="15px,15px,14px,14px"
  font-weight="500"
  tag="label"
  type="body"
>
  Some sort of text
</label>
`;

exports[`Typography should render line height correctly 1`] = `
.c0 {
  color: #1b2733;
  font-weight: 500;
  font-size: 15px;
  line-height: 24px;
  font-display: fallback;
}

@media screen and (min-width:400px) {
  .c0 {
    font-size: 15px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    font-size: 14px;
  }
}

@media screen and (min-width:786px) {
  .c0 {
    font-size: 14px;
  }
}

<label
  class="c0"
  color="text"
  font-size="15px,15px,14px,14px"
  font-weight="500"
  lineheight="24px"
  tag="label"
  type="body"
>
  Some sort of text
</label>
`;

exports[`Typography should render type correctly 1`] = `
.c0 {
  color: #1b2733;
  font-weight: 500;
  font-size: 15px;
  font-display: fallback;
}

@media screen and (min-width:400px) {
  .c0 {
    font-size: 15px;
  }
}

@media screen and (min-width:576px) {
  .c0 {
    font-size: 14px;
  }
}

@media screen and (min-width:786px) {
  .c0 {
    font-size: 14px;
  }
}

<label
  class="c0"
  color="text"
  font-size="15px,15px,14px,14px"
  font-weight="500"
  tag="label"
  type="body"
>
  Some sort of text
</label>
`;
